java/test/src/main/java/org/ray/api/test/ReferenceCountingTest.java

```diff
@@ -136,4 +138,20 @@ public void testDependencyRefCounts() {
     ((RayObjectImpl<?>) result).removeLocalReference();
     checkRefCounts(ImmutableMap.of());
   }
+
+  public void testBasicNestedIds() {
+    RayObject<byte[]> inner = Ray.put(new byte[40 * 1024 * 1024]);
+    RayObject<List<RayObject<byte[]>>> outer = Ray.put(Collections.singletonList(inner));
+
+    byte[] innerObjecrIdBytes = inner.getId().getBytes();
+    inner = null;
+    System.gc();
+
+    for (int i = 0; i < 100; i++) {
+      Ray.put(new byte[40 * 1024 * 1024]);
```

Comment:
The total size you put here is 4G, which seems too large for me. Can it be smaller? Maybe add some comments to explain this in the code.

Reply:
Modified to not rely on the behavior of GC.
Just check local reference to verify that ref counts are manipulated correctly.